src/application/view_photo.c

```diff
@@ -0,0 +1,465 @@
+    viewPhoto->lastInputWasTouch = sub_020183F0(&fieldSystem->unk_10C);
+    FieldViewPhoto_GetAlbumScrollParam(viewPhoto->parent, &viewPhoto->scrollData);
+    return SysTask_CreateOnMainQueue(SysTask_ViewPhoto, viewPhoto, 1);
+}
+
+void FieldSystem_DestroyViewPhotoTask(FieldSystem *fieldSystem) {
+    ViewPhotoSysTaskData *viewPhoto = (ViewPhotoSysTaskData *)SysTask_GetData(fieldSystem->unk_D8);
+
+    sub_02018410(&fieldSystem->unk_10C, viewPhoto->lastInputWasTouch);
+    ViewPhotoSysTask_Teardown(viewPhoto);
+    FreeToHeap(viewPhoto);
+    SysTask_Destroy(fieldSystem->unk_D8);
+    fieldSystem->unk_D8 = NULL;
+}
+
+static void SysTask_ViewPhoto(SysTask *task, void *taskData) {
+    ViewPhotoSysTaskData *viewPhoto = (ViewPhotoSysTaskData *)taskData;
+    switch (viewPhoto->state) {
+    case 0:
+        ViewPhotoSysTask_Setup(viewPhoto);
+        ++viewPhoto->state;
+        return;
+    case 1:
+        if (GX_GetMasterBrightness() == 0 && ViewPhotoSysTask_HandleInput(viewPhoto) != VIEW_PHOTO_INPUT_NOTHING) {
+            ++viewPhoto->state;
+        }
+        break;
+    case 2:
+        if (ViewPhotoSysTask_IsButtonAnimPlaying(viewPhoto)) {
+            FieldViewPhoto_SetPlayerInput(viewPhoto->parent, viewPhoto->lastInput);
+        }
+        break;
+    }
+    sub_0202457C(viewPhoto->spriteRender.spriteList);
+}
+
+static void ViewPhotoSysTask_Setup(ViewPhotoSysTaskData *viewPhoto) {
+    ViewPhotoSysTask_InitBgLayers(viewPhoto);
+    ViewPhotoSysTask_LoadBgGraphics(viewPhoto);
+    ViewPhotoSysTask_InitMessages(viewPhoto);
+    ViewPhotoSysTask_InitWindows(viewPhoto);
+    ViewPhotoSysTask_CreateSprites(viewPhoto);
+    ViewPhotoSysTask_PrintTextOnWindows(viewPhoto);
+}
+
+static void ViewPhotoSysTask_Teardown(ViewPhotoSysTaskData *viewPhoto) {
+    ViewPhotoSysTask_DeleteSprites(viewPhoto);
+    ViewPhotoSysTask_ReleaseWindows(viewPhoto);
+    ViewPhotoSysTask_ReleaseMessages(viewPhoto);
+    ViewPhotoSysTask_UnloadBgGraphics(viewPhoto);
+    ViewPhotoSysTask_ReleaseBgLayers(viewPhoto);
+}
+
+static ViewPhotoInputResponse ViewPhotoSysTask_HandleInput(ViewPhotoSysTaskData *viewPhoto) {
+    ViewPhotoInputResponse response = ViewPhotoSysTask_GetTouchInput(viewPhoto);
+    if (response == VIEW_PHOTO_INPUT_NOTHING) {
+        response = ViewPhotoSysTask_GetKeyInput(viewPhoto);
+        if (response == VIEW_PHOTO_INPUT_NOTHING) {
+            viewPhoto->lastInputWasTouch = FALSE;
+        }
+    } else {
+        viewPhoto->lastInputWasTouch = TRUE;
+    }
+    switch (response) {
+    case VIEW_PHOTO_INPUT_END:
+        ViewPhotoSysTask_AnimateButtonSelect(viewPhoto, 0);
+        PlaySE(SEQ_SE_DP_DECIDE);
+        viewPhoto->lastInput = VIEW_PHOTO_INPUT_END;
+        break;
+    case VIEW_PHOTO_INPUT_PREV:
+        if (viewPhoto->scrollData.curPhoto == 0) {
+            return VIEW_PHOTO_INPUT_NOTHING;
+        }
+        ViewPhotoSysTask_AnimateButtonSelect(viewPhoto, 1);
+        PlaySE(SEQ_SE_DP_SELECT);
```

Comment:
as a side note, can we come up with better SE consts, out of scope for this PR, but as much as I want it to be, this sadly is not DP, but HGSS

Reply:
the const comes straight from the sound data and should be considered immutable

Reply:
they could have better names for code facing stuff, sure they may be that in the sdat but we could do with some better const names, possibly two sets of consts if necessary, idk just throwing around ideas, I find some of the names very stupid